server/src/test/java/io/crate/integrationtests/InsertIntoIntegrationTest.java

```diff
@@ -1993,4 +1992,23 @@ ON CONFLICT (real_id, \"date\", year)
             "2030| 99998.0"
         );
     }
+
+    @Test
+    public void test_non_deterministic_sub_column_unused_in_targets() {
+        execute("""
+            CREATE TABLE tbl (
+                a int PRIMARY KEY,
+                o1 object as (
+                    sub int as round((random() + 1) * 100)
+                )
+            )
+            """
+        );
+
+        execute("INSERT INTO tbl(a) VALUES (1)");
+        refresh();
+        execute("SELECT o1['sub'] FROM tbl WHERE a = 1");
```

Comment:
should we execute this multiple times to make sure that primary + replica return same value?

Reply:
Good point, I think we need to test that (because we had multiple times that issue) and I faced it on another PR related to a different issue. Also, because of that I will keep integration test and won't replace it by a unit test.

Reply:
Added primary/replica comparison.